### PR TITLE
[1/4] Update the Rack SDK to support ruby 3.x

### PR DESCRIPTION
**Pull request by HzANut**
Ticket: https://www.pivotaltracker.com/n/projects/1646401/stories/181066034

Update the Rack SDK to support ruby 3.x

[2/4] Moesif/moesifapi-ruby#13
[3/4] Moesif/moesif-rack#38
[4/4] Moesif/moesif-rack-example#12

### CODE REVIEW

File: unirest.gemspec [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
   s.authors     = ["Moesif, Inc"]
   s.email       = 'derric@moesif.com'
   s.homepage    = 'https://github.com/Moesif/unirest-ruby'
   s.license     = 'MIT'
 
-	s.add_dependency('rest-client', '>= 1.8.0')
+  s.add_dependency('rest-client', '>= 1.8.0')
   s.add_dependency('json', '>= 1.8.1')
-	s.add_dependency('addressable', '>= 2.3.5')
+  s.add_dependency('addressable', '~> 2.8.0')
```
Comment (praves77) [inline unirest.gemspec]: Please make it `>= 2.8.0`

File: unirest.gemspec [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
-	s.add_development_dependency('shoulda', '~> 3.5.0')
-	s.add_development_dependency('test-unit')
-	s.add_development_dependency('rake')
+  s.add_development_dependency('shoulda', '~> 3.5.0')
+  s.add_development_dependency('test-unit')
```
Comment (praves77) [inline unirest.gemspec]: In this PR https://github.com/Moesif/moesifapi-ruby/pull/13/files `test-unit` has version dependency but not here. Let's have a consistency and add version here too.

**Comment by praves77** [comment]
@XinyuLuan left a few comments across all PRs. Let's make that change, test it then explore (spend ~1-2 hrs max) if it can work w/o adding `addressable`.

**Comment by keyur9** [comment]
lgtm

**Review by praves77 — APPROVED** [review]

**Pull request merged**